ecommerce/frontend/src/components/SignUpForm.js

```diff
@@ -0,0 +1,289 @@
+            <Form.Item
+                name="phone"
+                label="Phone Number"
+            >
+                <Input.Group compact >
+                    <Form.Item
+                        name={['phone', 'countrycode']}
+                        noStyle
+                        rules={[{ required: true, message: 'Please enter your country code!' }]}
+                    >
+                        <Input style={{ width: '20%' }} placeholder="+90" />
+                    </Form.Item>
+                    <Form.Item
+                        name={['phone', 'number']}
+                        noStyle
+                        rules={[{ required: true, message: 'Please enter your phone number!' }]}
+                    >
+                        <Input style={{ width: '80%' }} />
+                    </Form.Item>
+                </Input.Group>
+            </Form.Item>
+
+            {/* Captcha input
+            TODO: This input is non-functional */}
+            <Form.Item label="Captcha" extra="We must make sure that your are a human.">
+                <Row gutter={8}>
+                    <Col span={12}>
+                        <Form.Item
+                            name="captcha"
+                            noStyle
+                            rules={[
+                                {
+                                    required: true,
+                                    message: 'Please input the captcha you got!',
+                                },
+                            ]}
+                        >
+                            <Input />
+                        </Form.Item>
+                    </Col>
+                    <Col span={12}>
+                        <Button>Get captcha</Button>
+                    </Col>
+                </Row>
+            </Form.Item>
+
+            {/* User type selection */}
+            <Form.Item
+                name="usertype"
+                rules={[
+                    {
+                        required: true,
+                        message: "Please select your user type!"
+                    },
+                ]}
+                {...tailFormItemLayout}
+            >
+                <Radio.Group>
+                    <Radio value="customer">Register as customer</Radio>
+                    <Radio value="vendor">Register as vendor</Radio>
+                </Radio.Group>
+            </Form.Item>
+
+            {/* Checkbox for the user agreement */}
+            <Form.Item
+                name="agreement"
+                valuePropName="checked"
+                rules={[
+                    {
+                        validator: (_, value) =>
+                            value ? Promise.resolve() : Promise.reject('Should accept agreement'),
+                    },
+                ]}
+                {...tailFormItemLayout}
+            >
+                <Checkbox>
+                    I have read the <a href="">agreement</a>
+                </Checkbox>
```

Comment:
Note to self: need to either ask customer for some agreement document or just generate a generic one.